### PR TITLE
Add support for find commands for appointments and properties. 

### LINKED ISSUE #5: As a user, I want to be able to search based on keywords in the property description and title
so that I can easily find properties that suit my clients’ needs.

### PR DESCRIPTION
**Pull request by w-yuchen**
Added support for find commands for appointments and properties. New classes were created, but old address book classes are still kept as reference. Also added basic tests for find property and find appointment.

### CODE REVIEW

**Review by Cp-John — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FindAppointmentCommand.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/FindAppointmentCommand.java:12]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/commands/FindPropertyCommand.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.property.PropertyContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/commands/FindPropertyCommand.java:12]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.Arrays;
+
+import seedu.address.logic.commands.FindAppointmentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java:13]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+public class FindAppointmentCommandParser implements Parser<FindAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java:18]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Arrays;
+
+import seedu.address.logic.commands.FindAppointmentCommand;
+import seedu.address.logic.commands.FindPropertyCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.property.PropertyContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java:14]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+public class FindPropertyCommandParser implements Parser<FindPropertyCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java:19]: Rmb to update this javadoc

File: src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindPropertyCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindAppointmentCommand.MESSAGE_USAGE));
```
Comment (w2vgd) [inline src/main/java/seedu/address/logic/parser/FindPropertyCommandParser.java:27]: Should the message_usage be from `FindPropertyCommand` instead?

File: src/main/java/seedu/address/model/appointment/AppointmentContainsKeywordsPredicate.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.appointment;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/appointment/AppointmentContainsKeywordsPredicate.java:10]: Rmb to update this javadoc

File: src/main/java/seedu/address/model/property/PropertyContainsKeywordsPredicate.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.property;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
```
Comment (w2vgd) [inline src/main/java/seedu/address/model/property/PropertyContainsKeywordsPredicate.java:10]: Rmb to update this javadoc

**Review by w2vgd — COMMENTED** [review]
A few comments

**Pull request merged**